src/main/java/com/artipie/rpm/http/RpmSlice.java

```diff
@@ -33,17 +39,49 @@
 /**
  * Artipie {@link Slice} for RPM repository HTTP API.
  * @since 0.7
+ * @todo #213:30min Add integration tests for RpmSlice to validate
+ *  that it can answer requests coming from the rpm original client.
+ *  There should be at least one test without authentication and one
+ *  test with authentication, see https://stackoverflow.com/a/26852734/1723695
+ *  to configure the client side.
  */
 public final class RpmSlice extends Slice.Wrap {
 
     /**
-     * New RPM repository HTTP API.
-     * @param storage Storage
+     * Ctor.
+     * @param storage The storage.
      */
     public RpmSlice(final Storage storage) {
+        this(storage, Permissions.FREE, Identities.ANONYMOUS);
+    }
+
+    /**
+     * Ctor.
+     * @param storage The storage.
+     * @param perms Access permissions.
+     * @param auth Auth details.
+     */
+    public RpmSlice(final Storage storage, final Permissions perms, final Authentication auth) {
+        this(storage, perms, new BasicIdentities(auth));
+    }
+
+    /**
+     * Ctor.
+     * @param storage Storage
+     * @param perms Access permissions.
+     * @param users Concrete identities.
+     */
+    public RpmSlice(final Storage storage, final Permissions perms, final Identities users) {
         super(
             new SliceRoute(
-                new SliceRoute.Path(new RtRule.ByMethod(RqMethod.GET), new SliceDownload(storage))
+                new SliceRoute.Path(
+                    new RtRule.ByMethod(RqMethod.GET),
+                    new SliceAuth(
+                        new SliceDownload(storage),
+                        new Permission.ByName("download", perms),
```

Comment:
@victornoel Just a side note, do we have other permissions defined for `rpm-adapter` slices? Wouldn't be better if we use some structure to define them (like an enum for example)?

Reply:
@paulodamaso I'm not clear about that. Actually, I was telling @g4s8 in another PR (https://github.com/artipie/maven-adapter/pull/94#discussion_r436792091) that maybe actually all this permission/auth setup shouldn't be done in each of the adapter but instead directly at the using app (or artipie server) level in order not to copy paste the same thing everywhere...

Reply:
@victornoel @paulodamaso I don't think we can implement authorization and authentication on upper abstraction level, because some adapters uses different authentication algorithms: e.g. Maven, NPM, RPM uses HTTP-basic, but Docker, Gem has custom auth protocols. Same for authorization: each adapter may have different set of permissions, e.g. Maven potentially may have: `download`, `upload-release`, `upload-snapshot`, but it's not relevant for RPM repository. So we created common interfaces and implementation in http module (`SliceAuth`, `Permissions`, etc) to be used in adapters.